lib/bitrange.h

```diff
@@ -65,13 +65,31 @@ class JSONLoader;
 namespace BitRange {
 
 namespace Detail {
+#if defined(__GNUC__) || defined(__clang__)
+/**
+ * @return boolean indicating if @p value is a power of two.
+ *
+ * This function is declared constexpr to allow it to be used in tests to optimize code generation.
+ */
+constexpr bool isPowerOfTwo(int value) { return value && !(value & (value - 1)); }
+
+constexpr int log2ForPowerOfTwo(int value) { return sizeof(value) * 8 - __builtin_clz(value) - 1; }
```

Comment:
This explicitly assumes that `sizeof(value) == 4` as `__builtin_clz` accepts unsigned argument. Also for the clarity I'd change the argument to `unsigned`.

In fact, we can use abseil implementations of C++20 `<bit`> routines: `absl::countl_zero` and their friends. With FIXME to replace the implementation with standard ones.

Instead of `isPowerOfTwo ` above we could use `absl::has_single_bit`.

Even if we'd use our own implementations I'd use C++20 names to ensure we can just drop them in the future.

Reply:
I wish I'd known about the Abseil alternatives earlier: would have saved my attempting to roll my own equivalents to the STL functions that exist in C++20 and beyond.

Reply:
Yeah, sorry. I would probably need to add some short summary to docs of what is available....

Reply:
BTW, do you know of a more portable alternative to `__builtin_constant_p`?

Reply:
> BTW, do you know of a more portable alternative to `__builtin_constant_p`?

Unfortunately, not. I believe the question was raised few times, but there is no direct alternative. Maybe `if consteval` from C++23 is the closest alternative.